### PR TITLE
Locks Crawling behind a do_after of 1 second, modified by the number of missing limbs.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Before : 
https://streamable.com/yu7om
After : 
https://streamable.com/d3deg

### Reasoning
Crawling is, in its current form, almost a straight upgrade to walking as it shields you from lubbed floors, makes you harder to hit by certain projectiles, and is also a good camouflage/ambush tool, while still offering the same movement speed and the possibility to disarm spam.

Said movement speed is a bug, and entirely unintended by @MadmanMartian, original coder of crawling.

As such, this PR locks the act of crawling behind a do_after of 1 seconds, preventing you from spamming that mouse button to escape security.

An intended, but additional effect is that it also locks you in a direction. If you clicked on a tile, you'll crawl on that tile exactly one second after.

You can still crawl diagonally.

:cl:
- tweak: Crawling now requires you to stand still for exactly 1 seconds before moving. Missing/damaged limbs significantly increase that delay. Hyperzine counteracts those adverse effects. Additionally, once you've began to crawl in a certain direction, you can no longer change said direction.

### CODE REVIEW

**Comment by SonixApache** [comment]
How does having less limbs affect this one second?

**Comment by ShiftyRail** [comment]
As of yet, it doesn't, I'll see if I can make it affect you.

**Comment by Chazach** [comment]
my position is not very defensible
all I can say is I enjoyed sliding around while it lasted and don't think it broke that much or seemed viable as a combat tool

**Comment by ShiftyRail** [comment]
Maybe I could make it a genetic power, or some similar meme.

>seemed viable as a combat tool

As a combat tool, certainly not, as an escape tool, most definitively.

**Comment by Reviire** [comment]
Sounds like it'd be really obnoxious. Would be nicer if there was a cooldown after crawling. Just stun the player for 5 seconds :^)

**Comment by ShiftyRail** [comment]
You now crawl noticeably slower if you have broken/missing arms, as well as being a bit knocked down after you initial crawl.

Taking hyperzine mitigates this, to an extent, as there is a minimal delay of 1 second no matter what. This also applies to phazon speed.
Now, should this base delay be overridden or not ?

**Comment by Diavill** [comment]
ur getin metagrudeged kiddo

**Comment by ShiftyRail** [comment]
@Diavill even if I put a way to bypass this delay ? C'mon, give me ideas

**Comment by Diavill** [comment]
the most powerful part of crawling is being able to disarm people, in my opinion, the effectiveness of crawling as an escape mechanic is nice and all but it's extremely location based, e.g. crawling through flaps or to avoid taser fire. In the first example it's a smart move to use crawling that way, the second is outclassed by bangers. A madlad crawling at you with disarm intent is what's truly terrible, how are you gonna stop him? Click on him? He's already under you and has pushed you on the ground, your egun/baton/flash now his.
I just personally think crawling is goofy and hilarious and I love it, I'd rather it not be gimped to a slower version but I do understand the """""balance""""" issues

**Comment by ShiftyRail** [comment]
You can still disarm spam though.
I only added a do_after thing.

**Comment by Diavill** [comment]
Isn't the thing a process and doing anything else will interrupt it?

**Comment by small-boss** [comment]
how am i gonna have crawl duels if this gets added? cant do skillfull maneuvers if you have a one second delay on movement

**Comment by Blithering** [comment]
you do it the way spess god intended, with motorised wheelchairs and lances stolen from xenoarch

**Comment by CptWad** [comment]
what if I click two tiles at once, before I finish crawling?

**Comment by ShiftyRail** [comment]
@CptWad You'll crawl towards the first one you clicked, unless you switch active hand, which cancels both moves and allows you to rethink your plans.

**Comment by Dacendeth** [comment]
Could you make the m_strong mutation affect crawling time minimum? Currently that genetic power does very little.

File: code/_onclick/other_mobs.dm [hunk 1]
```diff
@@ -53,9 +53,11 @@
...
 
 	if(src.lying && !(isUnconscious() || stunned || paralysis) && check_crawl_ability() && isfloor(A) && isfloor(get_turf(src)) && proximity && !pulledby && !locked_to && !client.move_delayer.blocked())
-		var/crawldelay = movement_delay()*3
-		Move(A, get_dir(src,A), glide_size_override = crawldelay)
-		delayNextMove(crawldelay,additive=1)
+		var/delay = round(1 + base_movement_tally()/5) * 1 SECONDS
+		if (do_after(src, A, delay))
+			var/crawldelay = movement_delay()*3
+			Move(A, get_dir(src,A), glide_size_override = delay) // So that we're still smooth
+			delayNextMove(crawldelay,additive=1)
```
Comment (DamianX) [inline code/_onclick/other_mobs.dm]: Is there a reason to use `crawldelay` here now rather than the new `delay`?

Comment (ShiftyRail) [reply code/_onclick/other_mobs.dm]: Not really, now that I think of it, given that `movement_delay()` and `base_movement_tally()` aren't so different.

**Pull request merged**